### PR TITLE
RP102 - details field equals id validation

### PR DESCRIPTION
**Pull request by MLainer1**
## Related Issues
fixes:

## Description
Added the RP102 validation to the new validate

### CODE REVIEW

File: demisto_sdk/commands/content_graph/tests/test_data/indicator_type.json [hunk 1]
```diff
@@ -8,7 +8,7 @@
...
...|\\([-\\d\\p{L}\\+&@#\\/%=~_\\$\\?!\\-:,;\\\\\\*–—―\\|\\[\\]]*\\))*)*|((?i)((((25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.)+(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?))|(([\\[\\]\\p{L}\\dֿ]+\\[?\\.\\]?){1,3}[\\p{L}]+))(:[\\d]+)?\\/([-\\p{L}\\+&@#\\/%=~_\\$\\?!\\-:,;\\\\\\*–—―\\|\\.]+([\\d]*)?)+)|((?:(?:https?|ftps?|hxxps?|sftp):((\\/\\/)|(\\\\\\\\)))[\\d][\\S]+)|(((https?|ftps?|hxxps?|sftp):((\\/\\/)|(\\\\\\\\)))?(\\[)?(([0-9a-fA-F]{1,4}:){7,7}[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,7}:|([0-9a-fA-F]{1,4}:){1,6}:[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,5}(:[0-9a-fA-F]{1,4}){1,2}|([0-9a-fA-F]{1,4}:){1,4}(:[0-9a-fA-F]{1,4}){1,3}|([0-9a-fA-F]{1,4}:){1,3}(:[0-9a-fA-F]{1,4}){1,4}|([0-9a-fA-F]{1,4}:){1,2}(:[0-9a-fA-F]{1,4}){1,5}|[0-9a-fA-F]{1,4}:((:[0-9a-fA-F]{1,4}){1,6})|:((:[0-9a-fA-F]{1,4}){1,7}|:)|fe80:(:[0-9a-fA-F]{0,4}){0,4}%[0-9a-zA-Z]{1,}|::(ffff(:0{1,4}){0,1}:){0,1}((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\\.){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])|([0-9a-fA-F]{1,4}:){1,4}:((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\\.){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9]))(\\])?(:[\\d]+)?(((\\/|\\\\)[\\d]+(\\/|\\\\)[-\\d\\p{L}+&@'#\\/%=~_$!\\-:.;\\\\*–—―|\\[\\]]+)|((\\/|\\\\)[^\\d \\n\\t]+)))",
```
Comment (JudahSchwartz) [inline demisto_sdk/commands/content_graph/tests/test_data/indicator_type.json:10]: lets check this out why this isnt failing at the moment

**Review by JudahSchwartz — APPROVED** [review]
looks great! lets double check on this comment:

File: demisto_sdk/commands/content_graph/tests/test_data/indicator_type.json (hunk 1 above)
Comment (MLainer1) [reply demisto_sdk/commands/content_graph/tests/test_data/indicator_type.json:10]: The original validation only runs on new files

File: demisto_sdk/commands/validate/validators/RP_validators/RP102_details_field_equals_id.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+
+class DetailsFieldEqualsIdValidator(BaseValidator[ContentTypes]):
+    error_code = "RP102"
+    description = "Validate that the id and the details fields are equal"
+    error_message = "id and details fields are not equal."
+    rationale = "To align with the platform requirements."
+    related_field = "id"
+    is_auto_fixable = False
+    related_file_type = [RelatedFileType.JSON]
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/validators/RP_validators/RP102_details_field_equals_id.py:23]:
```suggestion
```

File: demisto_sdk/commands/validate/validators/RP_validators/RP102_details_field_equals_id.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    BaseValidator,
+    ValidationResult,
+)
+
+ContentTypes = IndicatorType
+
+
+class DetailsFieldEqualsIdValidator(BaseValidator[ContentTypes]):
+    error_code = "RP102"
+    description = "Validate that the id and the details fields are equal"
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/validators/RP_validators/RP102_details_field_equals_id.py:18]:
```suggestion
    description = "Validate that the id and the details fields are equal."
```

File: demisto_sdk/commands/validate/tests/RP_validatiors_test.py [hunk 4]
```diff
@@ -55,3 +58,29 @@ def test_ExpirationFieldIsNumericValidator_is_valid(
...
+    When
+        - Calling the DetailsFieldEqualsIdValidator is valid function.
+    Then
+        - Make sure the right amount of failures return. Should fail object two.
+    """
+    content_items = [
+        create_indicator_type_object(["details", "id"], ["test", "test"]),
+        create_indicator_type_object(["details", "id"], ["test", "test-not-equal"]),
+    ]
+    expected_msgs = ["id and details fields are not equal."]
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/tests/RP_validatiors_test.py:78]: Maybe let's mention the id and detail fields values?

**Review by YuvHayun — APPROVED** [review]
Nice!
Minor comments.
Approved for my side.

**Pull request merged**